src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,176 @@
+                                                        internalListener.onFailure(
+                                                            new FlowFrameworkException("Failed to update index setting for: " + indexName)
+                                                        );
+                                                    }
+                                                }, exception -> {
+                                                    logger.error("Failed to update index setting for: " + indexName, exception);
+                                                    internalListener.onFailure(exception);
+                                                }));
+                                        } else {
+                                            internalListener.onFailure(new FlowFrameworkException("Failed to update index: " + indexName));
+                                        }
+                                    }, exception -> {
+                                        logger.error("Failed to update index " + indexName, exception);
+                                        internalListener.onFailure(exception);
+                                    })
+                                );
+                        } else {
+                            // no need to update index if it does not exist or the version is already up-to-date.
+                            indexMappingUpdated.get(indexName).set(true);
+                            internalListener.onResponse(true);
+                        }
+                    }, e -> {
+                        logger.error("Failed to update index mapping", e);
+                        internalListener.onFailure(e);
+                    }));
+                } else {
+                    // No need to update index if it's already updated.
+                    internalListener.onResponse(true);
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Failed to init index " + indexName, e);
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Check if we should update index based on schema version.
+     * @param indexName index name
+     * @param newVersion new index mapping version
+     * @param listener action listener, if update index is needed, will pass true to its onResponse method
+     */
+    private void shouldUpdateIndex(String indexName, Integer newVersion, ActionListener<Boolean> listener) {
+        IndexMetadata indexMetaData = clusterService.state().getMetadata().indices().get(indexName);
+        if (indexMetaData == null) {
+            listener.onResponse(Boolean.FALSE);
+            return;
+        }
+        Integer oldVersion = NO_SCHEMA_VERSION;
+        Map<String, Object> indexMapping = indexMetaData.mapping().getSourceAsMap();
+        Object meta = indexMapping.get(META);
+        if (meta != null && meta instanceof Map) {
+            @SuppressWarnings("unchecked")
```

Comment:
Why do we have suppress warnings here?

Reply:
I suspect because all we know is that `meta` is an `Object` that an instance of `Map` but we don't know the types of its key and value when they are cast.